src/classpath/classpathConfigurationView.ts

```diff
@@ -109,16 +111,27 @@ async function initializeWebview(context: vscode.ExtensionContext): Promise<void
 async function checkRequirement(): Promise<boolean> {
     const javaExt = vscode.extensions.getExtension("redhat.java");
     if (!javaExt) {
-        // todo: check extension version
         classpathConfigurationPanel?.webview.postMessage({
             command: "onException",
             exception: ClasspathViewException.JavaExtensionNotInstalled,
         });
-        const err: Error = new Error("redhat.java is not installed or the version is too stale.");
+        const err: Error = new Error("The extension 'redhat.java' is not installed.");
         setUserError(err);
         sendError(err);
         return false;
     }
+
+    if (compareVersions(javaExt.packageJSON.version, LEAST_JLS_VERSION) < 0) {
+        classpathConfigurationPanel?.webview.postMessage({
+            command: "onException",
+            exception: ClasspathViewException.StaleJavaExtension,
+        });
+        const err: Error = new Error("The extension version of 'redhat.java' is too stale.");
```

Comment:
do you think we should also report current version if java extension?

Reply:
Yes I think so